hubble/src/github_fetcher/fetcher.rs

```diff
@@ -0,0 +1,142 @@
+use std::thread::sleep;
+
+use serde::Serialize;
+use sqlx::{Acquire, Postgres};
+use tracing::{debug, error, info, warn};
+
+use crate::github_fetcher::{
+    self,
+    client::{fetch_commit_details, fetch_file_contents, CommitDetails, FileDownloadError},
+    postgres::get_subscriptions,
+    Attempt, Download, Subscription,
+};
+
+#[derive(Debug, thiserror::Error)]
+pub enum UpdateSubscriptionError {
+    #[error("database error creation transaction for subscription {0}: {1}")]
```

Comment:
minor nit, non blocking: prefer to use `#[source]` instead of displaying the error (`sqlx::Error` in this case) as you will lose all sources other than the top level error